http.go

```diff
@@ -106,6 +116,45 @@ func InstrumentHandlerFunc(label string, f func(http.ResponseWriter, *http.Reque
 	return InstrumentHandler(label, http.HandlerFunc(f))
 }
 
+// traceSampler is a time-based request sampler for tracing.
+//
+// It allows at most one request to be traced during each time interval.
+type traceSampler struct {
+	interval time.Duration
+	rng      *rand.Rand
+
+	mu             sync.Mutex
+	nextSampleTime time.Time
+}
+
+// newTraceSampler returns a new traceSampler that allows at most one request
+// to be traced during each time interval.
+func newTraceSampler(interval time.Duration) *traceSampler {
+	now := time.Now()
+	return &traceSampler{
+		interval:       interval,
+		rng:            rand.New(rand.NewSource(now.UnixNano())),
+		nextSampleTime: now,
+	}
+}
+
+// shouldTrace returns true iff a request should be traced.
+func (s *traceSampler) shouldTrace() bool {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	now := time.Now()
+	if now.Before(s.nextSampleTime) {
+		return false
+	}
+
+	// Pick a new value for nextSampleTime and return.
```

Comment:
I suspect that unlike retry we can just randomly pick a duration in the range [0,2*period].

```
mult := rng.Float64(2*float64(s.interval))
```

(Retry was being clever by staying within [0.6,1.4] to ensure we don't retry too quickly, but here I think the simpler approach works)

Reply:
Nice, done.